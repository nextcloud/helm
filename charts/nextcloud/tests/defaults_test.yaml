suite: test my defaults
templates:
  - deployment.yaml
  - secrets.yaml
  - service.yaml
chart:
  version: 1.0.0
  appVersion: 1.2.3
tests:
  - it: defaults
    set:
      metrics.token: "abcd"
    asserts:
      - matchSnapshot: { }