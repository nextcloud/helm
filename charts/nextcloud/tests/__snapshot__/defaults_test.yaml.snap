defaults:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: nextcloud
        helm.sh/chart: nextcloud-1.0.0
      name: RELEASE-NAME-nextcloud
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: nextcloud
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            hooks-hash: 9525c2748a6c7cd0e28ec740623d0b3fa5a75c83b51ccfd136bc89c76737b204
            nextcloud-config-hash: 8266a725d5474acb6adbf9f0609a3494dc3340a3ac306db90eac9ddb1b851960
            php-config-hash: 44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: nextcloud
        spec:
          containers:
            - env:
                - name: SQLITE_DATABASE
                  value: nextcloud
                - name: NEXTCLOUD_ADMIN_USER
                  valueFrom:
                    secretKeyRef:
                      key: nextcloud-username
                      name: RELEASE-NAME-nextcloud
                - name: NEXTCLOUD_ADMIN_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: nextcloud-password
                      name: RELEASE-NAME-nextcloud
                - name: NEXTCLOUD_TRUSTED_DOMAINS
                  value: nextcloud.kube.home
                - name: NEXTCLOUD_DATA_DIR
                  value: /var/www/html/data
              image: nextcloud:1.2.3-apache
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  httpHeaders:
                    - name: Host
                      value: nextcloud.kube.home
                  path: /status.php
                  port: 80
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: nextcloud
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  httpHeaders:
                    - name: Host
                      value: nextcloud.kube.home
                  path: /status.php
                  port: 80
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              resources: {}
              volumeMounts:
                - mountPath: /var/www/
                  name: nextcloud-main
                  subPath: root
                - mountPath: /var/www/html
                  name: nextcloud-main
                  subPath: html
                - mountPath: /var/www/html/data
                  name: nextcloud-main
                  subPath: data
                - mountPath: /var/www/html/config
                  name: nextcloud-main
                  subPath: config
                - mountPath: /var/www/html/custom_apps
                  name: nextcloud-main
                  subPath: custom_apps
                - mountPath: /var/www/tmp
                  name: nextcloud-main
                  subPath: tmp
                - mountPath: /var/www/html/themes
                  name: nextcloud-main
                  subPath: themes
          securityContext:
            fsGroup: 33
          volumes:
            - emptyDir: {}
              name: nextcloud-main
  2: |
    apiVersion: v1
    data:
      nextcloud-password: Y2hhbmdlbWU=
      nextcloud-username: YWRtaW4=
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: nextcloud
        helm.sh/chart: nextcloud-1.0.0
      name: RELEASE-NAME-nextcloud
    type: Opaque
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: nextcloud
        helm.sh/chart: nextcloud-1.0.0
      name: RELEASE-NAME-nextcloud
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
          targetPort: 80
      selector:
        app.kubernetes.io/component: app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: nextcloud
      type: ClusterIP
